addons/web/static/src/core/datetime/datetime_picker.scss

```diff
@@ -16,85 +16,84 @@
+        
+        &:not(.o_select_start, .o_select_end) {
+            background: $o-component-active-bg;
+        }
     }
 
+    .o_current:not(.o_select_start,.o_select_end),
+    .o_highlight_start:not(.o_selected),
+    .o_highlight_end:not(.o_selected),
     .o_select_start,
     .o_select_end {
-        --bg-opacity: 1;
 
-        color: var(--white);
+        &::before {
+            content: "";
+            position: absolute;
+            box-shadow: inset 0 0 1px 1px #009688;
+            width: 100%;
+            aspect-ratio: 1;
+            border-radius: 100%;
+            z-index: 1;
+        }
     }
 
-    .o_select_start:not(.o_highlighted),
-    .o_highlight_start {
-        &.o_cell_md {
-            --DateTimePicker__Cell-border-radius-start: 0.3rem;
+
+    .o_select_start,
+    .o_select_end {
+        &:before {
+            background: #b6e6e2;
         }
-        &.o_cell_lg {
-            --DateTimePicker__Cell-border-radius-start: 0.5rem;
+
+        &:after {
+            content: "";
+            position: absolute;
+            background: $o-component-active-bg;
+            width: 50%;
+            aspect-ratio: 1/2;
+            z-index: -1;
         }
-    }
 
-    .o_select_end:not(.o_highlighted),
-    .o_highlight_end {
-        &.o_cell_md {
-            --DateTimePicker__Cell-border-radius-end: 0.3rem;
+        &:not(.o_select_end):after {
+            right: 0;
+            z-index: 0;
         }
-        &.o_cell_lg {
-            --DateTimePicker__Cell-border-radius-end: 0.5rem;
+
+        &:not(.o_select_start):after {
+            right: 50%;
+            z-index: 0;
         }
     }
 
-    .o_today:not(.o_out_of_range) {
-        --DateTimePicker__Today-border-color: var(--primary);
-        --DateTimePicker__Today-border-offset: 2px;
-        --DateTimePicker__Today-border-width: 5px;
-
-        color: var(--DateTimePicker__Today-border-color);
+    .o_today:not(.o_out_of_range) span {
         position: relative;
-        box-shadow: inset 0 0 0 var(--DateTimePicker__Today-border-offset)
-            var(--DateTimePicker__Today-border-color);
-        font-weight: bold;
-
-        &.o_selected,
-        &.o_current {
-            --DateTimePicker__Today-border-color: white;
-        }
 
-        &:before {
+        &::after {
             content: "";
-            right: var(--DateTimePicker__Today-border-offset);
-            bottom: var(--DateTimePicker__Today-border-offset);
-            border: solid transparent;
-            border-width: 0 0 var(--DateTimePicker__Today-border-width)
-                var(--DateTimePicker__Today-border-width);
-            border-bottom-color: var(--DateTimePicker__Today-border-color);
             position: absolute;
+            background: #CE151F;
+            width: .75em;
+            height: .5em;
+            bottom: $spacer * -.25;
+            border-radius: $border-radius-pill;
+            left: 50%;
+            transform: translateX(-50%);
```

Comment:
I suggest to design the dash a bit more slim
![image](https://github.com/odoo/odoo/assets/8558351/12c17c03-64d4-40ea-8fd2-8e30e0149c5e)

example, compiled CSS:
```css
    content: "";
    position: absolute;
    background: #CE151F;
    width: 0.95em;
    height: 0.2em;
    bottom: -3px;
    border-radius: 41.9rem;
    left: 50%;
    transform: translateX(-50%);
```